src/js/edu-benefits/pages/schoolSelection.js

```diff
@@ -41,11 +42,19 @@ export default function createSchoolSelectionPage(
   const pickFields = _.pick(fields);
 
   const schemaProperties = pickFields(schema.properties);
+
+  let requiredFields = [];
+  if (programRequired) {
```

Comment:
E.g. just like educationTypeRequired will cause the 'educationType' field to be required, we can have a function argument programNameRequired that will make the 'name' field required.